# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 7ea2efe7df94
# Full Hash: 7ea2efe7df94b7cfb09b2033fc4e28e2d6d4efa8
# Author: Ziran Sun <zsun@igalia.com>
# Date: 2023-05-20 09:21:40
# Regressor Bug: 1832837
# File Overlap Count: 1
# Description:
#   Bug 1832837 - Remove the fullscreen element ready check on fullscreen element descendant. r=edgar
#   
#   https://fullscreen.spec.whatwg.org/#fullscreen-element-ready-check
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D177913
# ==============================================================================

diff -r 846df2d0be58 -r 7ea2efe7df94 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Fri May 19 06:59:48 2023 +0000
+++ b/dom/base/Document.cpp	Fri May 19 10:14:37 2023 +0000
@@ -15286,15 +15286,6 @@
     aRequest.Reject("FullscreenDeniedHTMLDialog");
     return false;
   }
-  // XXXsmaug Note, we don't follow the latest fullscreen spec here.
-  //         This whole check could be probably removed.
-  if (fullscreenElement && !nsContentUtils::ContentIsHostIncludingDescendantOf(
-                               elem, fullscreenElement)) {
-    // If this document is fullscreen, only grant fullscreen requests from
-    // a descendant of the current fullscreen element.
-    aRequest.Reject("FullscreenDeniedNotDescendant");
-    return false;
-  }
   if (!nsContentUtils::IsChromeDoc(this) && !IsInFocusedTab(this)) {
     aRequest.Reject("FullscreenDeniedNotFocusedTab");
     return false;